### PR TITLE
OBJ Loader with test

### PR DESCRIPTION
**Pull request by Ahdhn**

### CODE REVIEW

File: include/RCubeViewer/RCubeViewer.h [hunk 1]
```diff
@@ -73,6 +73,8 @@ class RCubeViewer : public rcube::Window
                                int numSegments);
     EntityHandle addSurface(const std::string name, const MeshData &data);
 
+    EntityHandle addOBJ(const std::string name, const MeshData &data);
```
Comment (pradeep-pyro) [inline include/RCubeViewer/RCubeViewer.h]: Let's not have this as a separate method. We already have `addSurface()` method above, the user just needs to call `addSurface("some_name", loadOBJ(filename));`

File: test/HelloOBJ/HelloOBJ.cpp [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+    //Load obj file 
+    std::string input_obj_file = std::string(OBJ_RESOURCE_PATH) + "/" +                                 
+                                 "armadillo.obj";
+    MeshData mesh = rcube::loadOBJ(input_obj_file);
+
+    //make sure the mesh is OK    
+    //assert(mesh.valid());
+
+    //Add the loaded obj to the viewer
+    EntityHandle mesh_handle = viewer.addOBJ("OBJMesh", mesh);
```
Comment (pradeep-pyro) [inline test/HelloOBJ/HelloOBJ.cpp]: See my previous comment above, this line could be `EntityHandle mesh_handle = viewer.addSurface("OBJMesh", mesh);`

File: test/CMakeLists.txt [hunk 3]
```diff
@@ -3,3 +3,4 @@ cmake_minimum_required(VERSION 3.9)
 add_subdirectory(HelloSphere)
 add_subdirectory(HelloWorld)
 add_subdirectory(PBR)
+add_subdirectory(HelloOBJ)
```
Comment (pradeep-pyro) [inline test/CMakeLists.txt]: Maybe name the test `OBJMesh`?

**Review by pradeep-pyro — COMMENTED** [review]
Looks great! I added some comments.

File: src/RCube/Core/Graphics/MeshGen/Obj.cpp [hunk 4]
```diff
@@ -0,0 +1,220 @@
...
+#include "RCube/Core/Graphics/OpenGL/Mesh.h"
+#include <algorithm>
+#include <iterator>
+#include <locale>
+#include <sstream>
+#include <string>
+
+#ifndef MAX_LINE_LENGTH
+#define MAX_LINE_LENGTH 2048
+#endif
```
Comment (pradeep-pyro) [inline src/RCube/Core/Graphics/MeshGen/Obj.cpp]: Maybe use `constexpr int MAX_LINE_LENGTH = 2048;`

File: src/RCube/Core/Graphics/MeshGen/Obj.cpp [hunk 5]
```diff
@@ -0,0 +1,220 @@
...
+
+    std::for_each(file_extension.begin(), file_extension.end(),
+                  [](char &c) { c = std::tolower(c); });
+    if (file_extension != "obj")
+    {
+        throw std::runtime_error("loadOBJ:: only read OBJ files. Input file extension is \n" +
+                                 file_extension + " and file name is " + file_name);
+    }
+
+    FILE *Objfile = fopen(file_name.c_str(), "r");
```
Comment (pradeep-pyro) [inline src/RCube/Core/Graphics/MeshGen/Obj.cpp:38]: Minor (no need to fix now): we should use `fstream` here

**Comment by Ahdhn** [comment]
I addressed all the major comments. Please have a look!

**Review by pradeep-pyro — APPROVED** [review]
Great, thank you!

**Comment by pradeep-pyro** [comment]
Can you also upload the obj file used in the test?

**Comment by Ahdhn** [comment]
Checkout #16

**Comment by Ahdhn** [comment]
Checkout #16

**Pull request merged**